jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/impl/DB2SqlDialect.java

```diff
@@ -0,0 +1,406 @@
+	{
+        // If identifier contains double quotation marks ", it needs to be espaced by another double quotation mark. E.g. "a""b" is the identifier a"b in the db.
+        return "\"" + identifier.replace("\"", "\"\"") + "\"";
+	}
+
+	@Override
+	public String applyQuoteIfNeeded(String identifier)
+	{
+        // Quoted identifiers can contain any unicode char except dot (.).
+        // This is a simplified rule, which might cause that some identifiers are quoted although not needed
+        boolean isSimpleIdentifier = identifier.matches("^[A-Z][0-9A-Z_]*");
+        if (isSimpleIdentifier) {
+            return identifier;
+        } else {
+            return applyQuote(identifier);
+        }
+	}
+
+	@Override
+	public boolean requiresCatalogQualifiedTableNames(
+			SqlGenerationContext context)
+	{
+		//DB2 does not know catalogs
+		return false;
+	}
+
+	@Override
+	public boolean requiresSchemaQualifiedTableNames(
+			SqlGenerationContext context)
+	{
+		return true;
+	}
+	
+	@Override
+	public SqlGenerationVisitor getSqlGenerationVisitor(SqlGenerationContext context) {
+		return new DB2SqlGenerationVisitor(this, context);
+	}
+
+	@Override
+	public NullSorting getDefaultNullSorting()
+	{
+		//default db2 behaviour is to set nulls to the end of the result
+		return NullSorting.NULLS_SORTED_AT_END;
+	}
+
+	@Override
+	public String getStringLiteral(String value)
+	{
+        // Don't forget to escape single quote
+        return "'" + value.replace("'", "''") + "'";
+	}
+	
+	@Override
+	public DataType mapJdbcType(ResultSet cols) throws SQLException {
+		DataType colType = null;
+		int jdbcType = cols.getInt("DATA_TYPE");
+		
+		switch (jdbcType) {
+		case Types.CLOB:
+			colType = DataType.createVarChar(DataType.maxExasolVarcharSize, DataType.ExaCharset.UTF8);
+			break;
+		case 1111:
+			colType = DataType.createVarChar(DataType.maxExasolVarcharSize, DataType.ExaCharset.UTF8);
+			break;
+		// set timestamp to varchar as db2 offers a much higher precision than exasol
+		// however java timestamp only supports nanoseconds -> varchar(32)
+		case Types.TIMESTAMP:
+			colType = DataType.createVarChar(32, DataType.ExaCharset.UTF8);
+			break;
+			
+		// db2 driver always delivers UTF8 Characters no matter what encoding is specified for var + char data
+        case Types.VARCHAR:
+        case Types.NVARCHAR:
+        case Types.LONGVARCHAR:
+        case Types.CHAR:
+        case Types.NCHAR:
+        case Types.LONGNVARCHAR: {
+            int size = cols.getInt("COLUMN_SIZE");
+            DataType.ExaCharset charset =  DataType.ExaCharset.UTF8;
+            if (size <= DataType.maxExasolVarcharSize) {
+                colType = DataType.createVarChar(size, charset);
+            } else {
+                colType = DataType.createVarChar(DataType.maxExasolVarcharSize, charset);
+            }
+            break;
+        }
+			
+		// VARCHAR  and CHAR for bit data -> will be converted to hex string so we have to double the size
```

Comment:
this seems to be a mix of tabs and whitespaces - however, since we don't have any code style guide yet you are free to change it (we usually use 4 whitespaces), or keep it as it is and we will merge the pull request.